app/backend/src/clj/teet/cooperation/cooperation_db.clj

```diff
@@ -129,7 +130,37 @@
   (get-in (du/entity db application-id)
     [:cooperation.application/activity :db/id]))
 
+(defn applicaiton-3rd-party [db application-id]
+  (get-in (du/entity db application-id)
+    [:cooperation.3rd-party/_applications 0 :cooperation.3rd-party/name]))
+
 (defn response-project-id [db response-id]
   ;(def *args [db response-id])
   (get-in (du/entity db response-id)
           [:cooperation.application/_response 0 :cooperation.3rd-party/_applications 0 :cooperation.3rd-party/project :db/id]))
+
+(defn- is-date-after
+  "Check if the given date after number of days"
+  [deadline days]
+  (let [date-days-after (java.util.Date.
+                          (+ (System/currentTimeMillis)
+                            (* 1000 60 60 24 days)))]
+    (> 0 (. deadline compareTo date-days-after))))
+
+(defn responses-to-be-expired
+  "Returns all Applications with Responses to be expired
+  in the given number of days"
+  [db project-id days]
+  (let [
+        third-parties
+        (mapv first
+          (d/q '[:find (pull ?e [:db/id :cooperation.3rd-party/name])
+                 :where [?e :cooperation.3rd-party/project ?project]
+                 :in $ ?project] db project-id))]
+    (filter #(is-date-after (nth % 3) days)
+      (d/q '[:find ?third-party ?application ?date ?response-deadline
+             :where [?third-party :cooperation.3rd-party/applications ?application]
```

Comment:
Instead of filtering results, add where clause like
`[(< ?response-deadline ?days)]` and give the time as in parameter